src/components/timeline.js

```diff
@@ -0,0 +1,253 @@
+    const segmentsBarMinTop = timelineState.bounds.top + timelineBarHeight;
+    const segmentsBarMaxTop = timelineState.bounds.top + (timelineBarHeight * 2);
+
+    const isExceedingVerticalBounds = event.clientY > segmentsBarMaxTop || event.clientY < segmentsBarMinTop;
+    if (isExceedingVerticalBounds) {
+      handleSegmentsBarMouseLeave();
+      return;
+    }
+
+    const currentLeft = event.clientX - timelineState.bounds.left;
+    const isInBounds = newMarkerBounds.some(bounds => bounds.leftMin <= currentLeft && currentLeft <= bounds.leftMax);
+    setNewMarkerState({ left: currentLeft, isInBounds });
+  };
+
+  const updateStates = useCallback(() => {
+    const bounds = timelineRef.current.getBoundingClientRect();
+    const msToPxRatio = timelineRef.current.clientWidth / length;
+    const minSegmentLength = MIN_PART_DURATION_IN_MS * msToPxRatio;
+    const markers = parts.slice(1).map(part => ({ key: part.key, left: part.startTimecode * msToPxRatio }));
+
+    const segments = parts.map((part, index) => {
+      const segment = { key: part.key, title: part.title };
+
+      if (parts.length === 1) {
+        segment.width = bounds.width;
+        return segment;
+      }
+      if (index === 0) {
+        segment.width = markers[index].left;
+        return segment;
+      }
+      if (index === parts.length - 1) {
+        segment.width = bounds.width - markers[index - 1].left;
+        return segment;
+      }
+      segment.width = markers[index].left - markers[index - 1].left;
+      return segment;
+    });
+
+    const markerBounds = parts.map((part, index) => {
+      const nextPartStartTimecode = parts[index + 1]?.startTimecode || length;
+      const currentPartDuration = nextPartStartTimecode - part.startTimecode;
+
+      if (currentPartDuration <= MIN_PART_DURATION_IN_MS) {
+        return null;
+      }
+
+      let leftMin = (part.startTimecode * msToPxRatio) + minSegmentLength;
+      let leftMax = (nextPartStartTimecode * msToPxRatio) - minSegmentLength;
+      const thereIsAlmostAPixelLeft = (leftMax - leftMin) <= 0;
+
+      if (thereIsAlmostAPixelLeft) {
+        leftMin -= 0.5;
+        leftMax += 0.5;
+      }
+      return { leftMin, leftMax };
+    }).filter(bound => bound);
+
+    setNewMarkerBounds(markerBounds);
+    setTimelineState({ markers, segments, msToPxRatio, bounds, minSegmentLength });
+  }, [parts, length]);
+
+  useEffect(() => {
+    if (timelineRef.current) {
+      updateStates();
+    }
+  }, [timelineRef, updateStates]);
+
+  useEffect(() => {
+    window.addEventListener('resize', updateStates);
+    return () => {
+      window.removeEventListener('resize', updateStates);
+    };
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    if (!dragState || isTouchDevice()) {
+      return () => { };
+    }
+    window.addEventListener('mousemove', handleWindowMouseMove);
+    window.addEventListener('mouseup', handleWindowMouseUp);
```

Comment:
We should also remove this handler when the dragging is over, shouldn't we?

Reply:
Instead of removing the listener I had removed the whole line at some point by mistake 😂 Added back, thanks.